### PR TITLE
SI-7486 regression in implicit resolution.

### PR DESCRIPTION
**Pull request by paulp**
What a touchy beast the compiler is.

### CODE REVIEW

**Comment by paulp** [comment]
Review by @adriaanm

**Comment by adriaanm** [comment]
LGTM

File: src/compiler/scala/tools/nsc/typechecker/Implicits.scala [hunk 1]
```diff
@@ -996,11 +996,15 @@ trait Implicits {
           if (divergence || DivergentImplicitRecovery.sym != null) {
             if (settings.Xdivergence211.value) DivergingImplicitExpansionError(tree, pt, DivergentImplicitRecovery.sym)(context)
             else throw DivergentImplicit
-          } else invalidImplicits take 1 foreach { sym =>
-            def isSensibleAddendum = pt match {
+          }
+          else if (invalidImplicits.nonEmpty) {
+            val sym = invalidImplicits.head
+            // We don't even dare look if errors are being buffered
+            // !sym.hasFlag(LOCKED) is a hail mary between SI-2206 and SI-7486
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Implicits.scala:1003]: This isn't quite consistent with:

``` scala
def isCyclicOrErroneous =
      try sym.hasFlag(LOCKED) || containsError(tpe)
      catch { case _: CyclicReference => true }
```

I would be more comfortable if we also caught `CyclicReference` here.

**Comment by JamesIry** [comment]
@adriaanm @paulp should @retronym's post-merge comment be turned into an issue for 2.10.3?

**Comment by paulp** [comment]
Yeah, I'm on it, it just turned into a ridiculously larger thing than I'd intended. (The entire infer patch is an outgrowth of it!)

**Comment by adriaanm** [comment]
Have you discovered the TigeRex?

**Pull request merged**